File: docs/guides/scala/src/main/scala/setHeaderUserIDThenSaveObjects.scala
import scala.concurrent.Future
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Await
import scala.concurrent.duration.Duration

import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions

def setHeaderUserIDThenSaveObjects(): Future[Unit] = {
  val playlists: Seq[Map[String, Any]] = Seq() // Your records

  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  playlists.foreach { playlist =>
    val playlistUserID = playlist("userID").toString
    Await.result(
      client.saveObjects(
        indexName = "<YOUR_INDEX_NAME>",
        objects = playlists,
        waitForTasks = false,
        batchSize = 1000,
        requestOptions = Some(
          RequestOptions
            .builder()
            .withHeader("X-Algolia-User-ID", "playlistUserID")
            .build()
        )
      ),
      Duration(5, "sec")
    )
  }

  Future.unit
}

File: docs/guides/javascript/src/setHeaderUserIDThenSaveObjects.ts
import { algoliasearch } from 'algoliasearch';

const playlists: Record<string, any>[] = [
  /* Your records */
];

try {
  const client = algoliasearch('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

  playlists.forEach(async (playlist) => {
    const playlistUserID = playlist['userID'];
    await client.saveObjects(
      { indexName: 'indexName', objects: playlists, waitForTasks: false, batchSize: 1000 },
      {
        headers: { 'X-Algolia-User-ID': playlistUserID },
      },
    );
  });
} catch (e) {
  console.error(e);
}
